WordPress/wordpress-playground
# wordpress-develop Pull Request previewer

## What is this PR doing?

Support for previewing wordpress-develop Pull Requests.

<img width="1133" alt="CleanShot 2023-10-13 at 18 53 52@2x" src="https://github.com/WordPress/wordpress-playground/assets/205419/e30e9ac6-c99e-4e81-b68f-45e14116f6c9">

## Tell me more!

The previewer lives at https://playground.wordpress.net/wordpress.html and supports a few query parameters:

* `?pr=5481` – preview that PR
* `?url=/wp-admin/post-new.php` – load that URL in WordPress
* `?mode=seamless` – redirect to a full-screen Playground without the browser chrome and any other extra UI elements

For example, I could use the following URL to preview [PR 5511](https://github.com/WordPress/wordpress-develop/pull/5511) and immediately navigate to `/wp-admin/post-new.php?test=1`:

https://playground.wordpress.net/wordpress.html?pr=5511&url=/wp-admin/post-new.php?test=1

![CleanShot 2023-10-17 at 17 45 36@2x](https://github.com/WordPress/wordpress-playground/assets/205419/2b4ec998-d485-4780-ae75-865d38219b2a)

## Caveats

The downloaded WordPress bundle is ~45MB and takes a while to download without any visible progress information. Let's find a way to optimize it.

## Testing instructions

1. Go to http://localhost:5400/website-server/wordpress.html
2. Preview PR 5481 (no other PR works yet, see https://github.com/WordPress/wordpress-develop/pull/5481)
3. Confirm in wp-admin the WordPress version is 6.4 dev

## Screenshots

<img width="1286" alt="CleanShot 2023-10-13 at 18 58 36@2x" src="https://github.com/WordPress/wordpress-playground/assets/205419/ca25a714-4491-4f63-b82c-b2178a2828b5">

cc @spacedmonkey @fabiankaegy @dmsnell @jonathanbossenger @dawidurbanski
